onnxruntime/core/optimizer/optimizer_execution_frame.cc

```diff
@@ -0,0 +1,134 @@
+
+#include "core/common/common.h"
+#include "core/common/status.h"
+#include "core/common/logging/logging.h"
+#include "core/common/logging/macros.h"
+#include "core/framework/tensorprotoutils.h"
+#include "core/framework/data_types.h"
+#include "core/framework/mldata_type_utils.h"
+#include "core/framework/kernel_registry.h"
+#include "core/framework/fuse_nodes_funcs.h"
+#include "core/optimizer/optimizer_execution_frame.h"
+
+namespace onnxruntime {
+
+OptimizerExecutionFrame::Info::Info(const std::vector<const Node*>& nodes,
+                                    const InitializedTensorSet& initialized_tensor_set) {
+  // Create CPU execution provider
+  // For now, CPU execution provider will be created every time when initilizing Info.
+  // Later, it will be changed to pass by Info ctor.
+  cpu_execution_provider_ = std::make_unique<CPUExecutionProvider>(CPUExecutionProviderInfo());
+  allocator_ptr_ = cpu_execution_provider_->GetAllocator(device_id_, mem_type_);
+  ORT_ENFORCE(allocator_ptr_ != nullptr, "Failed to get allocator for optimizer");
+
+  // Create MLValues related maps
+  auto initialize_maps = [this, &initialized_tensor_set](const NodeArg* arg) -> Status {
+    int idx = mlvalue_name_idx_map_.Add(arg->Name());
+    mlvalue_idx_nodearg_map_[idx] = arg;
+
+    // Only create MLValue instances for initializers used by an array of nodes.
+    InitializedTensorSet::const_iterator it = initialized_tensor_set.find(arg->Name());
+    if (it != initialized_tensor_set.cend()) {
+      MLValue mlvalue;
+      utils::TensorProtoToMLValue(*(it->second), allocator_ptr_, nullptr, 0, mlvalue);
```

Comment:
>TensorProtoToMLValue [](start = 13, length = 20)

Not sure when the optimizer is running and what's available. Would the MLValue's for the initializers from the SessionState for the full Graph be available?

Reply:
It is running after the model is loaded and before SessionState is fully initialized. MLValues for the initializers from the SessionState are not available.